errors/error.go

```diff
@@ -4,78 +4,124 @@ import (
 	// the routing strategy routing strategy returns an empty routes list
-	ErrRouterStrategyReturnedEmptyRoutes = &HTTPError{
-		Code:    http.StatusNotImplemented,
-		Message: "fiber: routing strategy returned empty routes list",
+	ErrRouterStrategyReturnedEmptyRoutes = func(protocol protocol.Protocol) *FiberError {
+		statusCode := http.StatusNotImplemented
+		if protocol == "GRPC" {
+			statusCode = int(codes.Unimplemented)
+		}
+		return &FiberError{
+			Code:    statusCode,
+			Message: "fiber: routing strategy returned empty routes list",
+		}
 	}
 
-	// ErrServiceUnavailable is a HTTPError that's returned when
+	// ErrServiceUnavailable is a FiberError that's returned when
 	// none of the routes in the routing strategy return a valid response
-	ErrServiceUnavailable = &HTTPError{
-		Code:    http.StatusServiceUnavailable,
-		Message: "fiber: no responses received",
+	ErrServiceUnavailable = func(protocol protocol.Protocol) *FiberError {
+		statusCode := http.StatusServiceUnavailable
+		if protocol == "GRPC" {
+			statusCode = int(codes.Unavailable)
+		}
+		return &FiberError{
+			Code:    statusCode,
+			Message: "fiber: no responses received",
+		}
 	}
 
-	// ErrRequestTimeout is a HTTPError that's returned when
+	// ErrRequestTimeout is a FiberError that's returned when
 	// no response if received for a given HTTP request within the configured timeout
-	ErrRequestTimeout = &HTTPError{
-		Code:    http.StatusRequestTimeout,
-		Message: "fiber: failed to receive a response within configured timeout",
+	ErrRequestTimeout = func(protocol protocol.Protocol) *FiberError {
+		statusCode := http.StatusRequestTimeout
+		if protocol == "GRPC" {
+			statusCode = int(codes.DeadlineExceeded)
+		}
+		return &FiberError{
+			Code:    statusCode,
+			Message: "fiber: failed to receive a response within configured timeout",
+		}
 	}
 
-	// ErrReadRequestFailed is a HTTPError that's returned when a http request cannot
+	// ErrReadRequestFailed is a FiberError that's returned when a http request cannot
 	// be read successfully
-	ErrReadRequestFailed = func(err error) *HTTPError {
-		return &HTTPError{
-			Code:    http.StatusInternalServerError,
+	ErrReadRequestFailed = func(protocol protocol.Protocol, err error) *FiberError {
+		statusCode := http.StatusInternalServerError
+		if protocol == "GRPC" {
+			statusCode = int(codes.Internal)
+		}
+		return &FiberError{
+			Code:    statusCode,
 			Message: fmt.Sprintf("fiber: failed to read incoming request: %s", err.Error()),
 		}
 	}
 
 	// ErrRequestFailed is a generic error that is created when problems are encountered fulfilling
 	// a request
-	ErrRequestFailed = func(err error) *HTTPError {
-		return &HTTPError{
-			Code:    http.StatusInternalServerError,
+	ErrRequestFailed = func(protocol protocol.Protocol, err error) *FiberError {
+		statusCode := http.StatusInternalServerError
+		if protocol == "GRPC" {
+			statusCode = int(codes.Internal)
```

Comment:
I think we should propagate the original error code thrown by the route instead of translating it to Internal error. 
For example in grpc. dispatcher it should be
```go
	conn, err := grpc.DialContext(context.Background(), config.Endpoint, grpc.WithTransportCredentials(insecure.NewCredentials()))
	if err != nil {
		// if ok is false, unknown codes.Unknown and Status msg is returned in Status
		responseStatus, _ := status.FromError(err)
		return nil, fiberError.ErrRequestFailed(
			protocol.GRPC,
            responseStatus,
			errors.New("grpc dispatcher: "+responseStatus.String()))
	}

```

Reply:
nvm, I think it's properly done.